packages/dev-approuter/lib/helpers.js

```diff
@@ -0,0 +1,133 @@
+                throw new Error(`Duplicate dependency "${route.dependency}" found in file ${path.join(process.cwd(), configFile)}.`);
+            } else {
+                config.dependencyRoutes[`${route.dependency}`] = route;
+            }
+        }
+    });
+    return config;
+};
+
+/**
+ * Applies the `config.dependencyRoutes` to the `config.routes` and removes them.
+ * @param {Object} config - the approuter configuration including all `dependencyRoutes`.
+ * @returns {Object} config - the approuter configuration that can be used to start the approuter.
+ */
+const applyDependencyConfig = (config) => {
+    config.routes?.forEach(route => {
+        if (route.dependency) {
+            route = config.dependencyRoutes[route.dependency];
+        }
+    });
+    delete config.dependencyRoutes;
+    return config;
+};
+
+/**
+ * Adds a destination to `process.env.destinations` for a given module.
+ * If a destination with this `name === moduleId` already exists, no new destination will be created.
+ * @param {String} moduleId - the id of the module that a destination is created for.
+ * @param {Number} port - the port of the localhost that the destination should point to.
+ * @param {String} mountPath - the path the module was mounted to and the destination should point to.
+ */
+const addDestination = (moduleId, port, mountPath) => {
+    let destinations = [];
+    if (process.env.destinations) {
+        destinations = JSON.parse(process.env.destinations);
+    }
+
+    let url;
+    if (mountPath) {
+        url = `http://localhost:${process.env.PORT || 5000}${mountPath}`;
+    } else {
+        url = `http://localhost:${port}`;
+    }
+    
+    // only add new destination if it's not already provided
+    let destinationAlreadyExists;
+    destinations.forEach(destination => {
+        const lowerCaseDestination = {};
+        Object.keys(destination).forEach(key => {
+            lowerCaseDestination[key.toLowerCase()] = destination[key];
+        })
+        if (lowerCaseDestination.name === moduleId) {
+            destinationAlreadyExists = true;
+        }
+    })
+    if (!destinationAlreadyExists) {
+        destinations.push({
+            Name: moduleId,
+            Authentication: "NoAuthentication",
+            ProxyType: "Internet",
+            Type: "HTTP",
+            URL: url
+        });
+        process.env.destinations = JSON.stringify(destinations);
+    }
+};
+
+/**
+ * Configures the route for a given CAP module.
+ * @param {String} moduleId - the id of the module that the route should be configured for.
+ * @param {String[]} servicePaths - an array of service paths that the CAP module serves.
+ * @param {Object} route - the route that is to be configured.
+ * @returns {Object} the configured route.
+ */
+const configureCAPRoute = (moduleId, servicesPaths, route) => {
+    route.source = servicesPaths.map(path => { return path + "(.*)" }).join("|");
```

Comment:
better use string literal syntax to concat string

Reply:
Ok